packages/build-utils/test/unit.builds-and-routes-detector.test.ts

```diff
@@ -78,18 +78,62 @@ describe('Test `detectBuilders`', () => {
     expect(builders![0].use).toBe('@vercel/node');
     expect(builders![0].src).toBe('api/users.js');
     expect(builders![1].use).toBe('@vercel/static');
-    expect(builders![1].src).toBe('!{api/**,package.json}');
+    expect(builders![1].src).toBe('!{api/**,package.json,middleware.[jt]s}');
     expect(builders!.length).toBe(2);
     expect(errors).toBe(null);
   });
 
+  it('no package.json + no build + root-level "middleware.js"', async () => {
+    const files = ['middleware.js', 'index.html', 'web/middleware.js'];
+    const { builders, errors } = await detectBuilders(files);
+    expect(builders![0].use).toBe('@vercel/node');
+    expect(builders![0].src).toBe('middleware.js');
+    expect(builders![0].config?.middleware).toEqual(true);
+    expect(builders![1].use).toBe('@vercel/static');
+    expect(builders![1].src).toBe('!{api/**,package.json,middleware.[jt]s}');
```

Comment:
Can we add some tests to verify this glob pattern works as expected?

Reply:
Added a test using `vc build` in 8e51c9f6f7af409d45189c53f7ffb88678d7eef1

Reply:
Thanks! I was thinking some unit tests like `middleware.ts` and `middleware.js` both match but `middleware.jsx` doesnt match